cfn-resources/maintenance-window/cmd/resource/resource.go

```diff
@@ -204,22 +154,47 @@ func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler
 		return *handlerError, nil
 	}
 
-	var res *mongodbatlas.Response
-	res, err := client.MaintenanceWindows.Reset(context.Background(), *currentModel.ProjectId)
-
+	resp, err := client.AtlasV2.MaintenanceWindowsApi.ResetMaintenanceWindow(context.Background(), *currentModel.ProjectId).Execute()
 	if err != nil {
-		_, _ = logger.Warnf("Delete - error: %+v", err)
-		return progress_events.GetFailedEventByResponse(err.Error(), res.Response), nil
+		return progress_events.GetFailedEventByResponse(err.Error(), resp), nil
 	}
 
-	// Response
-	event := handler.ProgressEvent{
+	return handler.ProgressEvent{
 		OperationStatus: handler.Success,
 		Message:         "delete successful",
-	}
-	return event, nil
+	}, nil
 }
 
 func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
 	return handler.ProgressEvent{}, errors.New("not implemented: List")
 }
+
+func (m Model) toAtlasModel() admin.GroupMaintenanceWindow {
+	return admin.GroupMaintenanceWindow{
+		DayOfWeek:            *m.DayOfWeek,
+		HourOfDay:            *m.HourOfDay,
+		StartASAP:            m.StartASAP,
+		AutoDeferOnceEnabled: m.AutoDeferOnceEnabled,
+	}
+}
+
+func get(client *util.MongoDBClient, currentModel Model) (*admin.GroupMaintenanceWindow, *handler.ProgressEvent) {
+	maintenanceWindow, resp, err := client.AtlasV2.MaintenanceWindowsApi.GetMaintenanceWindow(context.Background(), *currentModel.ProjectId).Execute()
+	if err != nil {
+		_, _ = logger.Warnf("Read - error: %+v", err)
+		ev := progress_events.GetFailedEventByResponse(err.Error(), resp)
+		return nil, &ev
+	}
+
+	if isResponseEmpty(maintenanceWindow) {
+		_, _ = logger.Warnf("Read - resource is empty: %+v", err)
+		ev := progress_events.GetFailedEventByCode("resource not found", cloudformation.HandlerErrorCodeNotFound)
+		return nil, &ev
+	}
+
+	return maintenanceWindow, nil
+}
+
+func isResponseEmpty(maintenanceWindow *admin.GroupMaintenanceWindow) bool {
+	return maintenanceWindow != nil && maintenanceWindow.DayOfWeek == 0
```

Comment:
is there a case where it'll be nil? it should only be nil if the method returns an error

Reply:
to be honest I kept the original logic just in case. I tend to think like you but don't know if this was for some edge case that I can't think of now.